src/wren/Overlay.cpp

```diff
@@ -59,31 +60,12 @@ namespace wren {
       mParams.mBackgroundColor = color;
   }
 
-  void Overlay::setTexture(Texture *texture) {
-    mTexture = texture;
-
-    if (mTexture)
-      mParams.mActiveFlags.y = 1.0f;
-    else
-      mParams.mActiveFlags.y = 0.0f;
-  }
-
-  void Overlay::setBackgroundTexture(Texture *texture) {
-    mBackgroundTexture = texture;
-
-    if (mBackgroundTexture)
-      mParams.mActiveFlags.x = 1.0f;
+  void Overlay::setTexture(Texture *texture, int role) {
+    mTextures[role] = texture;
+    if (texture)
+      mParams.mActiveFlags.x = mParams.mActiveFlags.x | (1u << role);
     else
-      mParams.mActiveFlags.x = 0.0f;
-  }
-
-  void Overlay::setForegroundTexture(Texture *texture) {
-    mForegroundTexture = texture;
-
-    if (mForegroundTexture)
-      mParams.mActiveFlags.z = 1.0f;
-    else
-      mParams.mActiveFlags.z = 0.0f;
+      mParams.mActiveFlags.x = mParams.mActiveFlags.x & ~(1u << role);
```

Comment:
It looks like it should be:
```suggestion
      mParams.mActiveFlags.y = mParams.mActiveFlags.y & ~(1u << role);
```
Isn't it?

Reply:
No, the code is correct.
The `mParams.mActiveFlags.y` contains if the border is active or not.
All the textures flags are stored in `mParams.mActiveFlags.x` value.